Use dynamic quantum architecture (DQA) as transpilation target (#124)
iqm-finland/qiskit-on-iqm: docs/user_guide.rst

   `Inspecting circuits before submitting them for execution`_ for inspecting the actual run request sent for
   execution.

The calibration data for an IQM quantum computer is stored in a calibration set. An :class:`.IQMBackend` instance
always corresponds to a specific calibration set, so that its transpilation target uses only those QPU components
(qubits and computational resonators) and gates which are available in that calibration set. The server default
calibration set will be used by default, but one can also use a different calibration set by specifying the
``calibration_set_id`` parameter for :meth:`.IQMProvider.get_backend` or :class:`.IQMBackend`.

> 

Author: Added explanation.

You can optionally set IQM backend specific options as additional keyword arguments to
:meth:`.IQMBackend.run`, documented at :meth:`.IQMBackend.create_run_request`.
For example, you can enable heralding measurements by passing the appropriate circuit compilation option as follows:

.. code-block:: python

    from iqm.iqm_client import CircuitCompilationOptions

>     job = backend.run(transpiled_circuit, shots=1000, circuit_compilation_options=CircuitCompilationOptions(heralding_mode=HeraldingMode.ZEROS))

Author: `backend.run()` now always uses the calibration set of the `backend`, so using the latest one may require creating a new `backend`. A warning is shown in that case, see `"Server default calibration set has changed ..."` warning in `IQMBackend.create_run_request`, which suggests to use a new backend. I'll also add a mention about this in the user guide as suggested in your other comment above.

   `Inspecting circuits before submitting them for execution`_ for inspecting the actual run request sent for
   execution.

The calibration data for an IQM quantum computer is stored in a calibration set. An :class:`.IQMBackend` instance
always corresponds to a specific calibration set, so that its transpilation target uses only those QPU components
(qubits and computational resonators) and gates which are available in that calibration set. The server default
calibration set will be used by default, but you can also use a different calibration set by specifying the
``calibration_set_id`` parameter for :meth:`.IQMProvider.get_backend` or :class:`.IQMBackend`. If the server default

> calibration set has changed after you created the backend, the backend would still use the original default calibration
> set when submitting circuits. In this case, a warning will be shown to notify you about this, and you would need to
> create a new backend if you want to use the newer calibration set.

Reviewer: ```suggestion
calibration set has changed after you have created the backend, the backend will still use the original default calibration
set when submitting circuits for execution. In this case, a warning will notify you about this.
You will need to create a new backend if you want to use the new default calibration set.
```

Author: Sorry, merged already, can be done in https://github.com/iqm-finland/qiskit-on-iqm/pull/123 I guess.

Reviewer: Applied
